Improve error message for unknown types in postgres source

chaas opened the pull request:
Update error message for unknown types in postgres source to specify which table/column are the culprit. Add detail and hint messages.
As per conversation in #15502 and the new style guidelines in https://github.com/MaterializeInc/materialize/pull/15631.

Example:
```
materialize=> CREATE SOURCE pg_source FROM POSTGRES CONNECTION pg_connection2 (PUBLICATION 'mz_source') FOR ALL TABLES WITH (SIZE = '3xsmall');
ERROR:  column "person.current_mood" uses unrecognized type
DETAIL:  type with OID 211538 is unknown
HINT:  You may be using an unsupported type in Materialize, such as an enum. Try excluding the table from the publication.
```

Required follow up: add testdrive support for hints and details.


### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way) and therefore is tagged with a `T-proto` label.
- [x] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):

Add hint to "type with OID x is unknown" errors for postgres sources, indicating the table and column of the unsupported type.

philip-stoev reviewed (approved):
Thank you. Seems to work as advertised.  I also manually tried `MONEY` and `tsrange` and the error+hint is raised appropriatlely.

chaas commented on src/sql/src/plan/error.rs:
```diff
@@ -114,6 +120,9 @@ impl PlanError {
                 Some("The prefix \"mz_\" is reserved for system timelines.".into())
             }
+            Self::UnknownTypeInPostgresSource { table, column , e: _ }  => {
+                Some(format!("You may be using an unsupported type in Materialize for column \"{}\" in table \"{}\".", column, table))
```
Neat!

chaas commented on src/testdrive/src/action/sql.rs:
```diff
@@ -392,6 +392,10 @@ async fn try_run_fail_sql(
             Some(err) => {
                 let mut err_string = err.message().to_string();
+                if let Some(hint) = err.hint() {
+                    err_string.push_str("\nhint: ");
```
Surprisingly it didn't, but I'm going to move it out anyways.

As per our newly formalized error standards https://github.com/MaterializeInc/materialize/pull/15631, I should use both a detail and a hint with this error, so I'll separate out the test framework changes to support both.

chaas commented:
FYI I updated the error message a bit, to be consistent with our newly formalized error standards https://github.com/MaterializeInc/materialize/pull/15631.

I also removed the testdrive change, and will tackle that in a separate PR, b/c now we have both details and hints.

philip-stoev reviewed (approved):
Thank you for the test!
